unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/PrivateChatEntry.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PrivateChatEntry : BaseComponentView
+{
+    [SerializeField] private Button openChatButton;
+    [SerializeField] private PrivateChatEntryModel model;
+    [SerializeField] private TMP_Text userNameLabel;
+    [SerializeField] private TMP_Text lastMessageLabel;
+    [SerializeField] private ImageComponentView picture;
+    [SerializeField] private UnreadNotificationBadge unreadNotifications;
+    [SerializeField] private Button optionsButton;
+    [SerializeField] private GameObject blockedContainer;
+    [SerializeField] private GameObject onlineStatusContainer;
+    [SerializeField] private GameObject offlineStatusContainer;
+    [SerializeField] private RectTransform userContextMenuPositionReference;
+
+    private UserContextMenu userContextMenu;
+    private IChatController chatController;
+    private ILastReadMessagesService lastReadMessagesService;
+
+    public PrivateChatEntryModel Model => model;
+
+    public event Action OnOpenChat;
+
+    public override void Awake()
+    {
+        base.Awake();
+        optionsButton.onClick.AddListener(() =>
+        {
+            userContextMenu.Show(model.userId);
+            Dock(userContextMenu);
+        });
+        openChatButton.onClick.AddListener(() => OnOpenChat?.Invoke());
+    }
+
+    public void Initialize(IChatController chatController,
+        UserContextMenu userContextMenu,
+        ILastReadMessagesService lastReadMessagesService)
+    {
+        this.chatController = chatController;
+        this.userContextMenu = userContextMenu;
+        this.lastReadMessagesService = lastReadMessagesService;
+        userContextMenu.OnBlock -= HandleUserBlocked;
+        userContextMenu.OnBlock += HandleUserBlocked;
+    }
+
+    public void Set(PrivateChatEntryModel model)
+    {
+        this.model = model;
+        RefreshControl();
+    }
```

Comment:
You can implement the interface `IComponentModelConfig` and replace this function by the `Configure(BaseComponentModel newModel)` one provided.

Reply:
done!